api/v1_projects.inc

```diff
@@ -799,6 +799,40 @@ function api_v1_project_validatetext($method, $data, array $query_params)
     return $text_checker->analyse(receive_project_text_from_request_body('text'), $data[":projectid"]);
 }
 
+function api_v1_project_pickersets($method, $data, $query_params)
+{
+    // expand the character ranges so that each character can have its name
+    $project = $data[":projectid"];
+    return expand_pickersets($project->get_pickersets());
+}
+
+function expand_pickersets($pickersets)
```

Comment:
Lets put most of this functionality in the PickerSet class, maybe a `get_verbose_subsets()` (or a better-named) function, and then move the API-specific bits like mapping to an array into `api_v1_project_pickersets()` directly.

Reply:
Done something like this.